pkg/services/ngalert/schedule/jitter.go

```diff
@@ -0,0 +1,68 @@
+package schedule
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/grafana/grafana/pkg/services/featuremgmt"
+	ngmodels "github.com/grafana/grafana/pkg/services/ngalert/models"
+	"github.com/prometheus/prometheus/model/labels"
+)
+
+// JitterStrategy represents a modifier to alert rule timing that affects how evaluations are distributed.
+type JitterStrategy int
+
+const (
+	JitterNever JitterStrategy = iota
+	JitterByGroup
+	JitterByRule
+)
+
+// JitterStrategyFromToggles returns the JitterStrategy indicated by the current Grafana feature toggles.
+func JitterStrategyFromToggles(toggles featuremgmt.FeatureToggles) JitterStrategy {
+	strategy := JitterNever
+	if toggles.IsEnabledGlobally(featuremgmt.FlagJitterAlertRules) {
```

Comment:
Is it possible to set both toggles at once? Should we return an error if this happens?

Reply:
It'll prioritize jitter by rule.

I think what we have is what we want, too. We'll probably enable the per-group one globally through an HG toggle, then enable the per-rule one on top for specific problem customers. The HG pod shouldn't crash in this case, it should just jitter by rule.

Reply:
It is, jitter by rule will win, and I think this is desirable. Especially with how we are positioning this with jitter by rule being an additional escape hatch.

Given cloud's feature toggle admin page (which we might make this togglable in), I don't want to give a path where someone can create invalid config.